Publish year in review blog post (#109)
bpmn-io/bpmn.io: src/blog/2021/006-2021-year-in-review.md

---

title: 2021 in Review
layout: blogpost.hbs

slug: 2021-year-in-review
author: Nico Rehwaldt <https://github.com/nikku>

published: 2021-12-21 12:00

releases:

---

<p class="introduction">
  2021, [form-js](https://bpmn.io/toolkit/form-js/) joined our bpmn.io family. Besides that, we shipped numerous modeling and stability improvements in our core projects. As always, exciting things happend on the side-line, too. Great times to look back at what we have accomplished! Good times to take a small glimpse into the near 2022 future, too.
</p>

<!-- continue -->

<div class="figure condensed-size">
  <img src="{{ assets }}/attachments/blog/2018/014-christmas-tree.gif" alt="Merry Christmas to whom it may concern!" style="border-color: #14966396">
  <p class="caption">
    It is that time of the year again!
  </p>
</div>

## The Year in Features

> Early this year [form-js](https://bpmn.io/toolkit/form-js/) joined the bpmn.io project family. With four follow up minor releases this represents our minimal take on for editing and embedding, powered by bpmn.io. With form-js we acknowledge the fact that, along with BPMN and DMN, forms are an [integral part of process communication](https://bpmn.io/blog/posts/2021-form-js-visual-form-editing-and-embedding.html).

Reviewer: To make it easier to understand, consider

```suggestion
Early this year [form-js](https://bpmn.io/toolkit/form-js/) joined the bpmn.io project family. It also retrieved four follow up minor releases and with those, represents our minimal take on form editing and embedding, powered by bpmn.io. With form-js we acknowledge the fact that, along with BPMN and DMN, forms are an [integral part of process communication](https://bpmn.io/blog/posts/2021-form-js-visual-form-editing-and-embedding.html).
```

Other greater topics were modernizations of our code-base, improvement of standards compatibility, and modeling UX improvements. Some noteworthy core library milestones accomplished this year are the following:

* Released in January, [`bpmn-js@8`](https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md#800) significantly improves mouse and keyboard navigation.
* In March, [form-js](https://bpmn.io/toolkit/form-js/) joins as our take on visual form editing and embedding.
* Released in May, [`bpmn-js@8.5`](https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md#850) significantly improves the message flow editing experience.

> * Released in June, [form-js@0.1](./2021-form-js-visual-form-editing-and-embedding.html) is our first stable form library release.

Reviewer 2: ```suggestion
* Released in June, [`form-js@0.1`](./2021-form-js-visual-form-editing-and-embedding.html) is our first stable form library release.
```

* Also released in June, [`bpmn-js@8.7`](https://github.com/bpmn-io/bpmn-js/blob/develop/CHANGELOG.md#870) ships native support for the BPMN in color specification. Along with that we ship a reworked [color picker extension](https://github.com/bpmn-io/bpmn-js-color-picker).

> * Also released in June, `dmn-js@v11.0` ships with [modern import and export APIs](http://localhost:9012/blog/posts/2021-dmn-js-11-0-awaitable-import-export-apis.html).

Reviewer: Why no link to dmn-js?

Author: :+1:

> * Also released in June, [`dmn-js@v11.0`](https://github.com/bpmn-io/dmn-js/blob/develop/packages/dmn-js/CHANGELOG.md#1100) ships with [modern import and export APIs](./2021-dmn-js-11-0-awaitable-import-export-apis.html).

Reviewer 2: ```suggestion
* Also released in June, [`dmn-js@11`](https://github.com/bpmn-io/dmn-js/blob/develop/packages/dmn-js/CHANGELOG.md#1100) ships with [modern import and export APIs](./2021-dmn-js-11-0-awaitable-import-export-apis.html).
```

Significant work happened in one of the many underlying libraries that support our editing ecosystem. As an example, we [published 20 releases](https://www.npmjs.com/package/diagram-js?activeTab=versions) of [diagram-js](https://github.com/bpmn-io/diagram-js/blob/develop/CHANGELOG.md), each improving the way our core modeling UX works.

We cut more than 95 releases (more than 40 being actual feature releases) in [over 30 libraries](https://github.com/bpmn-io). To communicate the new and noteworthy in our ecosystem, we published [five blog posts](https://bpmn.io/blog/) and sent out [numerous tweets](https://twitter.com/bpmn_io).

## The Year in Numbers

> Similar to [previous](./2018-year-in-review.html) [years](./2019-year-in-review.html) we see significant increase in interest in our project and toolkits.

Reviewer: ```suggestion
Similar to [previous](./2018-year-in-review.html) [years](./2019-year-in-review.html) we were able to significantly increase the interest in our projects and toolkits.
```

> More people visited [our project website](https://bpmn.io/), asked for help, and contributed to [our forum](https://forum.bpmn.io/) and starred our projects on [GitHub](https://github.com/bpmn-io). Community work led to several contributions, too, such as [spanish translations for bpmn-js](https://github.com/bpmn-io/bpmn-js-i18n/pull/11) and [improved](https://github.com/bpmn-io/diagram-js/pull/484) [keyboard accessibility](https://github.com/bpmn-io/diagram-js/pull/493) of our editors.

Reviewer: ```suggestion
More people visited [our project website](https://bpmn.io/), asked for help, contributed to [our forum](https://forum.bpmn.io/), and starred our projects on [GitHub](https://github.com/bpmn-io). Community work led to several contributions, too, such as [spanish translations for bpmn-js](https://github.com/bpmn-io/bpmn-js-i18n/pull/11) and [improved](https://github.com/bpmn-io/diagram-js/pull/484) [keyboard accessibility](https://github.com/bpmn-io/diagram-js/pull/493) of our editors.
```

<div class="figure no-border condensed-size" style="margin: 50px 0 40px 0; text-align: center">

>   <img src="{{ assets }}/attachments/blog/2021/006-stats.png" alt="Some project statistics" style="max-width: 720px">

Reviewer: Consider to make the increased bundle size grey in colour

Author: Why?

  <p class="caption" style="margin-top: 10px">
    2021 in numbers (with percentages compared to 2020).
  </p>
</div>

Newly implemented features paid a minimal toll with regards to bundle size. If you'd like that number to shrink, [roll your own, minimal BPMN editor](https://github.com/bpmn-io/bpmn-js/pull/1252).

## The Year as a Collaborative Effort

With two more folks on board in 2021, we are now at a respectable team of seven developers, dedicated to working on our toolkits as well as the core Camunda modeling apps. Head over to [our about page](https://bpmn.io/about/) to learn more about us.

## Looking Ahead <a name="looking-ahead"></a>

Making great BPM tooling available everywhere for everyone is a mission, not a task on a checklist. Considering this, our future could not be more exciting. So what is up next?

> In 2022 we are going to ship a full rework of our properties panel infrastructure, and new properties panels for BPMN, DMN, and forms. Async data loading will be easily possible. Promised <img class="emoji" src="https://twemoji.maxcdn.com/v/13.1.0/svg/1f91e-1f3fc.svg" alt="crossed fingers">.

Reviewer: Consider link to bpmn-js-properties-panel
